app/code/Magento/InventoryInStorePickup/Model/ExtractSourceShippingAddressData.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickup\Model;
+
+use Magento\Framework\DataObject\Copy;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\InventoryInStorePickupShippingApi\Model\Carrier\InStorePickup;
+use Magento\Quote\Api\Data\AddressInterface;
+
+/**
+ * Extract Shipping Address fields from Source.
+ */
+class ExtractSourceShippingAddressData
```

Comment:
Service should extract address data. Not Shipping Address Data.
Shipping Address should be extracted in Quote module.